VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioMesh.cpp

```diff
@@ -81,10 +81,13 @@ void FVitruvioMesh::Build(const FString& Name, TMap<Vitruvio::FMaterialAttribute
 	TArray<FTriIndices> Indices;
 	const auto PolygonGroups = MeshDescription.PolygonGroups();
 	size_t MaterialIndex = 0;
+
+	TMap<FString, int32> UniqueMaterialNames;

```

Comment:
It wasn't an issue before but was certainly confusing. The create material function did the uniquify as well if clashes happened. I made some changes to the code anways:

Now we also (similar to the instance replacements) select materials by an "identifier". Before it could happen, that two materials (where the user did not set a material name in CGA and the name was created from the color map, or if the CGA material name was the same between materials) clashed and would have been postfixed by numbers. This is also not deterministic (which number does the material get?) which would result in wrong replacements. Now if two materials have the same "identifier" they will be replaced simultanously (eg both of these materials would be replaced all the time).